README_cmake.md

```diff
@@ -304,17 +695,219 @@ and the `USE_RUNTIME` option is ignored. Other options work as expected.
 
 If `REGISTRATION` is set, the path to the generated `.registration.cpp` file
 will be set in `OUTVAR`. This can be used to generate a runner for a Halide
-library that is useful for benchmarking and testing. For example:
-
-```
-add_halide_library(my_lib FROM my_gen REGISTRATION REG_CPP)
-add_executable(my_gen_runner ${REG_CPP})
-target_link_libraries(my_gen_runner PRIVATE my_lib Halide::RunGenMain)
-```
-
-This is equivalent to setting `-e registration`.
+library that is useful for benchmarking and testing, as documented above. This
+is equivalent to setting `-e registration` at the generator command line.
 
 Lastly, each of the `extra-output` arguments directly correspond to an extra
 output (via `-e`) from the generator. The value `OUTVAR` names a variable into
 which a path (relative to `CMAKE_CURRENT_BINARY_DIR`) to the extra file will be
 written.
+
+## Cross compiling
+
+Cross-compiling is a fraught topic in CMake when some binaries must be built for
```

Comment:
nit: 'fraught' is a scary word here and the wording is maybe a bit awkward. Maybe: "Cross-compiling in CMake can be tricky, since CMake doesn't easily support compiling for both the host platform and the cross-platform within the same build"?

Reply:
Missed this earlier. Used your wording.